database/README.md

```diff
@@ -0,0 +1,118 @@
+# Database refresh
+
+Use-cases:
+
+1. Show the whole workflow to the user.
+
+- It's not clear what it is.
+
+2. For each step, we get:
+
+- previous step
+- next steps
+- other steps from this group (e.g. other chroots for this build)
+
+4. It is possible to rerun the whole workflow.
+5. It is possible to rerun one step (and all the follow-up steps).
+6. It is possible to rerun a part of one step (and the follow-up step(s)).
+
+- E.g. one chroot.
+
+7. For project, we get all workflows.
+8. For project, we get all events.
+9. For event, we get all workflows.
+10. For event, we get a project.
+11. For each step, we get event.
+12. For each step, we get project.
+
+## How to deal with chroots?
+
+Have a grouping model for all chroots that are related. In Copr, all the chroots are build together and we can group those builds together, but where the grouping does not exists, we can do it manually (=implement the logic on our side).
+
+![](./img/copr_runs.png)
+
+## How to do pipelines?
+
+We can create a new `Step` model that will be a middle point between `Pipeline` and build/test model. The `Step` model can also track the relation to other `Steps`. One test/build can be connected to multiple `Steps`:
+
+- No `n:m` mapping between `Pipeline` and test/build model is needed.
+- Sharing of test/build is possible if we retrigger part of the workflow.
+
+![](./img/erd.png)
+
+We just need to make sure that there is only one build/test item linked to one `Step`:
+
+![](./img/erd_only_one.png)
+
+Here are the queries we need to do:
+
+![](./img/erd_how_to_get.png)
+
+The following two images show the `Pipeline` from the object point of view.
+
+The whole pipeline retriggered:
+
+![](./img/object_model_2.png)
+
+Only build retriggered:
+
+![](./img/object_model_1.png)
+
+### Check-runs UX
+
+Currently, user can request a rerun for a single chroot using the `retry` button. With that, there will be a new pipeline for each chroot we re-trigger. To be able to group all the retried chroots, we can create a [so-called requested action](https://docs.github.com/en/rest/guides/getting-started-with-the-checks-api#check-runs-and-requested-actions) to re-trigger all or all failed. We can also provide choise of the step to re-trigger.
+
+## Effectiveness of the queries
+
+For the regular queries covering multiple entries that goes across the database, it might be better to create them as one query and not relly on the ORM.
+
+### Migrations
+
+- For the model that groups the Copr chroots together, we need to group them by `copr_id` or `SRPMBuildModel`.
+- In case of `TF`, `SRPMBuildModel` is probably the only way, but this doesn't cover the test retrigger. We can use `TFTTestRunModel.submitted_time`.
```

Comment:
could you please include that for tests without the build phase the `SRPMBuildModel` cannot be used as well?